src/tree-internal-api.ts

```diff
@@ -0,0 +1,48 @@
+import { TreeService } from './tree.service';
+import { Tree } from './tree';
+import { TreeModel } from './tree.types';
+import { TreeInternalComponent } from './tree-internal.component';
+import { Subject, Observable } from 'rxjs/Rx';
+import { Injectable, Inject, ElementRef } from '@angular/core';
+import * as _ from 'lodash';
+
+export class TreeInternalAPI {
+  constructor (private treeInternalComponent: TreeInternalComponent) { }
+
+  public select(e: MouseEvent = new MouseEvent('click')): void {
+    if (this.treeInternalComponent && !this.treeInternalComponent['isSelected'] && typeof this.treeInternalComponent['onNodeSelected'] === 'function') {
+      this.treeInternalComponent['onNodeSelected'](e);
+    }
+  }
+
+  public switchFoldingType(): void {
+    if (this.treeInternalComponent && typeof this.treeInternalComponent['onSwitchFoldingType'] === 'function') {
+      this.treeInternalComponent['onSwitchFoldingType']();
+    }
+  }
+
+  public expand(): void {
+    if (this.treeInternalComponent && this.treeInternalComponent['tree'] instanceof Tree) {
+      if (!this.treeInternalComponent['tree']['isNodeExpanded'].call(this.treeInternalComponent['tree'])) {
+        this.switchFoldingType();
+      }
+    }
+  }
+
+  public collapse(): void {
+    if (this.treeInternalComponent && this.treeInternalComponent['tree'] instanceof Tree) {
+      if (this.treeInternalComponent['tree']['isNodeExpanded'].call(this.treeInternalComponent['tree'])) {
+        this.switchFoldingType();
+      }
+    }
+  }
+
+  public rename(newValue: string): void {
+    if (this.treeInternalComponent && this.treeInternalComponent['tree'] instanceof Tree) {
+      this.treeInternalComponent['tree']['value'] = newValue;
+    }
+  }
+
+  public reloadChildren(): void {
```

Comment:
I'm not sure how to implement this right now - I wanted to discuss it with you.
I need a way to call the load children function again to update a node's children.